packages/next-app/src/app/metadata/(components)/RootLayout.tsx

```diff
@@ -23,18 +24,19 @@ import { OwnedPool } from "./OwnedPool";
 export function RootLayout({ children }: { children: React.ReactNode }) {
   const { chain } = useNetwork();
   const { isConnected } = useAccount();
-  const router = useRouter();
 
   const RainbowKitProviderDynamic = dynamic(
     async () => (await import("@rainbow-me/rainbowkit")).RainbowKitProvider,
     { ssr: false }
   );
 
-  useEffect(() => {
-    localStorage.setItem("networkId", chain?.id?.toString() || "1");
+  let { address } = useAccount();
 
-    router.refresh();
-  }, [chain]);
+  address = impersonateWhetherDAO(chain?.id.toString() || "1", address);
+
+  const { data } = gql(chain?.id.toString() || "1").usePool({
+    owner: address,
+  });
 
   return (
     <WagmiConfig client={client}>
```

Comment:
Wonder if we should split this layout so that it could be reused by the admin app.